app/src/main/java/com/example/moviesapp/fragment/NowPlayingFragment.kt

```diff
@@ -5,11 +5,25 @@ import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.GridLayoutManager
+import com.example.domain.entity.Movie
+import com.example.moviesapp.R
+import com.example.moviesapp.adapter.NowPlayingAdapter
 import com.example.moviesapp.databinding.FragmentNowPlayingBinding
+import com.example.moviesapp.util.Constants
+import com.example.moviesapp.viewmodel.NowPlayingViewModel
+import org.koin.android.ext.android.inject
 
 class NowPlayingFragment : Fragment() {
 
     private lateinit var binding: FragmentNowPlayingBinding
+    private val viewModel: NowPlayingViewModel by inject()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        viewModel.getMovieState().observe({ lifecycle }, ::updateUI)
```

Comment:
these two lines should be called in the OnViewCreated method not in the onCreate

Reply:
done